### DISCUSSION TITLE
No file with "__main__" found in workspace directory

### DISCUSSION THREAD

**Discussion by katiepaff** [community/community#123819] [New to GitHub]
Hi all, I'm new to using this platform and I was wondering what it means if you are trying to run and calculate the grade for your assignment(Cengage mindtap) and you get the error message listed in the title? 

I tried changing the file I was working on to just __main__.py, but that didn't help

**Comment by CleverLemming1337** [comment]
Hello @katiepaff,
Could you please give some more information like what files you have and what you exactly want to do?

**Reply by DevClaros** [reply]
I have similar questions, and well we have 3 files ( calculate_gains_overtime.py, calculate_gains.py,  functions_optional.py) 
![image](https://github.com/community/community/assets/172628753/6e471485-82d2-47b2-8563-03d4206e742a)

I'm unsure it's an error on the Cengage Mindtap companion and files.

**Comment by DevClaros** [comment]
So I just found out the solution to this by putting the following code: 

if __name__ == "__main__":
 
typically this error occurs in GitHub Codespaces when it tries to launch your application but cannot find the entry point it expects to run.

**Reply by egulino** [reply]
Where do you put this code? I have the same issue.

**Reply by CleverLemming1337** [reply]
> So I just found out the solution to this by putting the following code:
> 
> if **name** == "**main**":
> 
> typically this error occurs in GitHub Codespaces when it tries to launch your application but cannot find the entry point it expects to run.

Hi, @DevClaros,

did you mean `if name == "__main__"`?

That‘s a common solution to execute code only when the file is manually executed, not imported by other files (because importing just means executing).

**Reply by egulino** [reply]
I was able to find it. I believe the line is:
if __name__ =="__main__": 
(typed with dunder underscores)

I typed it just after the import statements, before the main(). I was able to get Cengage to run my code.
